crates/pallet-messenger/src/relayer.rs

```diff
@@ -76,4 +78,48 @@ impl<T: Config> Pallet<T> {
 
         Ok(relayer_id)
     }
+
+    /// Unreserve the deposit and remove relayer to the relayer set.
+    /// Also adjust the next relayer index so that we wont skip any relayer due to relayer exit.
+    pub(crate) fn do_exit_relayer_set(
+        caller: T::AccountId,
+        relayer_id: RelayerId<T>,
+    ) -> DispatchResult {
+        // ensure relayer is in the set.
+        let relayer = RelayersInfo::<T>::take(relayer_id.clone()).ok_or(Error::<T>::NotRelayer)?;
+
+        // ensure caller is the owner of the relayer
+        ensure!(relayer.owner == caller, Error::<T>::NotOwner);
+
+        // release the deposit
+        T::Currency::unreserve(&caller, T::RelayerDeposit::get());
```

Comment:
This should be `relayer.deposit_reserved` as `T::RelayerDeposit` may be changed since the registration.

Reply:
good catch!